File: LeetCode/Easy/819. Most Common Word.py
def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
    paragraph = paragraph.lower()  # 소문자로 변경
    words = re.sub('[\'?!;",.]',' ',paragraph).split()      # 구두점 제거 + 공란 기준으로 split
    words = [word for word in words if word not in banned]  # banned 단어 필터링
    words = collections.Counter(words)  # Counter 객체로 최빈값 계산
    return words.most_common(1)[0][0]   # 최빈값 리턴

File: LeetCode/819_Most Common Word.swift
class Solution {
    func mostCommonWord(_ paragraph: String, _ banned: [String]) -> String {
        var wordCheckDict = [String: Int]()

        // 단어 처리(소문자 + 공백 변환) & 구분(공백 기준)
        let words = paragraph.lowercased().replacingOccurrences(of: ",", with: " ").split(separator: " ")

        // 단어 반복하면서 갯수 체크
        for word in words {
            var newWord = word.components(separatedBy: ["!", "?", "'", ",", ";", "."]).joined()
            // 딕셔너리에 등록된 word가 아닌 경우 + banned 단어가 아닌 경우 -> 새롭게 딕셔너리 카운트 추가
            if (wordCheckDict[newWord] == nil && !banned.contains(newWord)) { 
                wordCheckDict[newWord] = 1 
            } else {
                if let count = wordCheckDict[newWord] { wordCheckDict[newWord] = count + 1 }
            }
        }

        // 가장 자주 등장하는 단어 찾기
        var mostCommonWord = ""
        var maxCount = 0
        for (word, count) in wordCheckDict {
            if count > maxCount {
                mostCommonWord = word
                maxCount = count
            }
        }
        
        return mostCommonWord
    }
}
